include/ozo/binary_deserialization.h

```diff
@@ -232,4 +232,23 @@ basic_result<T>& recv_result(basic_result<T>& in, const oid_map_t<M>&, basic_res
     return out;
 }
 
+template <typename T, typename M>
+basic_result<T>& recv_result(basic_result<T>& in, const oid_map_t<M>& oid_map, std::reference_wrapper<basic_result<T>> out) {
+    return recv_result(in, oid_map, out.get());
+}
+
+template <typename M>
+basic_result<shared_native_result_handle>& recv_result(basic_result<native_result_handle>& in,
```

Comment:
Do you really see an application for the shared result? I mean, we provide the result as-is only if a user wants to do something non-standard with raw things. It's ok, but is it really needed to provide more functionality in this direction?

Reply:
If you need create handler with only result, you need to make `shared_ptr` wrapper over `ozo::result`. `ozo::shared_result` allow not to do it, and use only one pointer.

Reply:
Removed